app/ante/fees_test.go

```diff
@@ -0,0 +1,300 @@
+package ante_test
+
+import (
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/tharsis/ethermint/app/ante"
+	"github.com/tharsis/ethermint/tests"
+	evmtypes "github.com/tharsis/ethermint/x/evm/types"
+)
+
+var execTypes = []struct {
+	name      string
+	isCheckTx bool
+	simulate  bool
+}{
+	{"deliverTx", false, false},
+	{"deliverTxSimulate", false, true},
+}
+
+func (s AnteTestSuite) TestMinGasPriceDecorator() {
+	denom := evmtypes.DefaultEVMDenom
+	testMsg := banktypes.MsgSend{
+		FromAddress: "evmos1x8fhpj9nmhqk8z9kpgjt95ck2xwyue0ptzkucp",
+		ToAddress:   "evmos1dx67l23hz9l0k9hcher8xz04uj7wf3yu26l2yn",
+		Amount:      sdk.Coins{sdk.Coin{Amount: sdk.NewInt(10), Denom: denom}},
+	}
+
+	testCases := []struct {
+		name     string
+		malleate func() sdk.Tx
+		expPass  bool
+		errMsg   string
+	}{
+		{
+			"invalid cosmos tx type",
+			func() sdk.Tx {
+				return &invalidTx{}
+			},
+			false,
+			"must be a FeeTx",
+		},
+		{
+			"valid cosmos tx with MinGasPrices = 0, gasPrice = 0",
```

Comment:
can you add another test with `MinGasPrices = 0, gasPrice > 0`?

Reply:
Done in https://github.com/tharsis/ethermint/pull/1104/commits/586d141de37479eac06d58224a6c601cfb70a776